plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py

```diff
@@ -60,19 +59,20 @@ def change_location_tag(primary_xml_snippet, relpath):
     Transform the <location> tag to strip out leading directories and add `Packages/<first_letter>`.
 
     :param primary_xml_snippet: snippet of primary xml text for a single package
-    :type  primary_xml_snippet: unicode
+    :type  primary_xml_snippet: str
 
     :param relpath: Package's 'relativepath'
-    :type  relpath: unicode
+    :type  relpath: str
     """
-
+    primary_xml_snippet = primary_xml_snippet.decode('utf-8', 'replace')
     start_index = primary_xml_snippet.find("<location ")
     end_index = primary_xml_snippet.find("/>", start_index) + 2  # adjust to end of closing tag
 
     first_portion = primary_xml_snippet[:start_index]
     end_portion = primary_xml_snippet[end_index:]
     location = """<location href="%s"/>""" % file_utils.make_packages_relative_path(relpath)
-    return first_portion + location + end_portion
+    modified_primary_xml_snippet = first_portion + location + end_portion
+    return modified_primary_xml_snippet.encode('utf-8')
```

Comment:
This part is not 100% necessary because there are places where we check if the xml snippet is in unicode and encode it then, but for the consistency [here](https://github.com/goosemania/pulp_rpm/blob/8252c281a02d9728791256d4f40dee392a46a7ff/plugins/pulp_rpm/plugins/importers/yum/parse/rpm.py#L49-L53) I think it is better to return all the metadata in the same way.